source/addonStore/models/status.py

```diff
@@ -479,6 +487,15 @@ def isPendingEnable(self) -> bool:
 	def isPendingDisable(self) -> bool:
 		return self.name in self._stateHandler[AddonStateCategory.PENDING_DISABLE]
 
+	@property
+	def _anyPendingInstallForId(self) -> bool:
+		from ..dataManager import addonDataManager
+		return (
+			Path(self.pendingInstallPath).exists()
+			or self.name in (d.model.name for d, _ in addonDataManager._downloadsPendingInstall)
+			or self.name in (d.model.name for d in addonDataManager._downloadsPendingCompletion)

```

Comment:
should this be rearranged for speed? is the path check slower than the iteration?

Reply:
I guess the path chck would probably be slower, though depends on how big the iterables are. I doubt it would make much difference either way.